src/context-menu/index.tsx

```diff
@@ -0,0 +1,91 @@
+import { create, tsx } from '@dojo/framework/core/vdom';
+import { createICacheMiddleware } from '@dojo/framework/core/middleware/icache';
+import theme from '../middleware/theme';
+import Menu, { MenuOption } from '../menu/index';
+import * as menuCss from '../theme/default/menu.m.css';
+import * as css from '../theme/default/context-menu.m.css';
+import * as fixedCss from './context-menu.m.css';
+
+export interface ContextMenuProperties {
+	options: MenuOption[];
+	onValue(value: string): void;
+	menuWidth?: number;
+}
+
+interface ContextMenuICache {
+	open: boolean;
+	x: number;
+	y: number;
+}
+
+const icache = createICacheMiddleware<ContextMenuICache>();
+const factory = create({ icache, theme }).properties<ContextMenuProperties>();
+
+export const ContextMenu = factory(function({
+	properties,
+	children,
+	middleware: { icache, theme }
+}) {
+	function onClose() {
+		icache.set('open', false);
+	}
+	const { options, onValue, menuWidth, theme: themeProp } = properties();
+	const open = icache.getOrSet('open', false);
+	const x = icache.getOrSet('x', 0);
+	const y = icache.getOrSet('y', 0);
+	const classes = theme.classes(css);
+	const wrapperStyles: { top: string; left: string; width?: string } = {
+		top: `${y}px`,
+		left: `${x}px`
+	};
+	if (typeof menuWidth !== 'undefined') {
+		wrapperStyles.width = `${menuWidth}px`;
+	}
+
+	return (
+		<virtual>
+			<div
+				key="content"
+				oncontextmenu={(event: MouseEvent) => {
+					event.preventDefault();
+					icache.set('x', event.clientX - 2);
+					icache.set('y', event.clientY - 4);
+					icache.set('open', true);
+				}}
+			>
+				{children()}
+			</div>
+			{open && (
+				<body>
+					<div key="underlay" classes={fixedCss.underlay} onclick={onClose} />
+					<div
+						key="wrapper"
+						classes={[fixedCss.root, classes.wrapper]}
+						styles={wrapperStyles}
+					>
+						<Menu
+							key="menu"
+							theme={{
```

Comment:
bear in mind this will need to change after https://github.com/dojo/widgets/pull/940

Reply:
👍 I believe I've updated it accordingly